test/put.sh

```diff
@@ -293,6 +293,33 @@ it_can_put_and_set_git_config() {
   mv ~/.gitconfig.orig ~/.gitconfig
 }
 
+it_can_put_and_force_the_push() {
+  local repo1=$(init_repo)
+
+  local src=$(mktemp -d $TMPDIR/put-src.XXXXXX)
+  local repo2=$src/repo
+  git clone $repo1 $repo2
+
+  local ref=$(make_commit $repo2)
+  make_commit $repo1
+  # cannot push to repo while it's checked out to a branch
+  git -C $repo1 checkout refs/heads/master
```

Comment:
We thought that is what is happening. We are making a commit on both sides and should lose the commit in repo1. Would checking that the commit made in repo1 is gone in repo2 satisfy this concern?